android-client-sdk/src/main/java/com/devcycle/sdk/android/api/EventQueue.kt

```diff
@@ -99,22 +103,39 @@ internal class EventQueue constructor(
             }
         }
 
-        if (eventQueue.size > 0 || !result.success) {
-            scheduler.scheduleWithDelay { run() }
+        if ((eventQueue.size > 0 || !result.success) && !isClosed.get()) {
+            scheduleJob = scheduler.scheduleWithDelay { run() }
         }
 
+        if (isClosed.get() && !flushMutex.isLocked){
```

Comment:
whats the purpose of checking the lock here?

Reply:
It's so that we don't prematurely call back with an error. The queue could be closed while the code is running inside the lock, which could potentially trigger `flushEvents` to be called again unnecessarily. If the lock is locked, then this code is about to get triggered again anyway

Reply:
basically we dont want the close callback to get called if the lock is occupied because that would mean its technically not done closing yet

Reply:
could you just put this whole thing inside the `withLock` context then?

Reply:
but the `result` Im relying on is set in so many different places inside that context. Like if an error happens early it will get missed

Reply:
but you can just do it at the end can't you? it would be the same as where it is now just inside the lock context

Reply:
Yeah I guess I can just call it after the catch